### ISSUE TITLE
Do we need a mask tensor for averaging

### ISSUE THREAD

**Issue by cxncu001** [brightmart/text_classification#39]
# 1.get emebedding of words in the sentence
        sentence_embeddings = tf.nn.embedding_lookup(self.Embedding,self.sentence)  #  [None,self.sentence_len,self.embed_size]
 # 2.average vectors, to get representation of the sentence
        self.sentence_embeddings = tf.reduce_mean(sentence_embeddings, axis=1)  # [None,self.embed_size]

Since the length of sentences is variable,  do we need a mask tensor for geting the average?
For example,  tf.reduce_sum(tf.multiply(sentence_embeddings,mask), axis=1) / tf.reduce_sum(mask, axis=1)

**Comment by brightmart** [comment]
no need.
e.g. if a sequence has three elements:{5.1,5.0,4.9}. another sequence has four elements:{4.8,4,9,5.1,5.2}. average of first one is 5.0, another of second is also 6.0.
AVERAGE is total/sequence_length.

**Comment by cxncu001** [comment]
When training, we pad all sentences as max sentence length,  then tf.reduce_mean(sentence_embeddings, axis=1) means sum/(max sentence length).   How could it know the length of each sentence?
